# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1519673630 21600
#      Mon Feb 26 13:33:50 2018 -0600
# Node ID b0be07092078260e4e25301843b5f02ef5ffe7b3
# Parent  1cacfebaed3e702be573dbe87fa5c34f0488682b
Remove unused hidden axes properties for camera view (bug #52940).

* graphics.in.h (axes::properties): Discard the hidden properties
  x_viewtransform, x_projectiontransform, x_viewporttransform,
  x_normrendertransform and x_rendertransform that appear to be no longer
  in use anywhere in the project.

* graphics.cc (axes::properties::update_camera): No longer set the local
  variables x_viewtransform, x_projectiontransform, x_viewporttransform,
  x_normrendertransform and x_rendertransform.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -5738,22 +5738,16 @@ axes::properties::update_camera (void)
   x_zlim(0) = cmin(2);
   x_zlim(1) = cmax(2);
 
   x_render = x_normrender;
   scale (x_render, xd/(xlimits(1)-xlimits(0)), yd/(ylimits(1)-ylimits(0)),
          zd/(zlimits(1)-zlimits(0)));
   translate (x_render, -xo, -yo, -zo);
 
-  x_viewtransform = x_view;
-  x_projectiontransform = x_projection;
-  x_viewporttransform = x_viewport;
-  x_normrendertransform = x_normrender;
-  x_rendertransform = x_render;
-
   x_render_inv = x_render.inverse ();
 
   // Note: these matrices are a slight modified version of the regular matrices,
   // more suited for OpenGL rendering (x_gl_mat1 => light => x_gl_mat2)
   x_gl_mat1 = x_view;
   scale (x_gl_mat1, xd/(xlimits(1)-xlimits(0)), yd/(ylimits(1)-ylimits(0)),
          zd/(zlimits(1)-zlimits(0)));
   translate (x_gl_mat1, -xo, -yo, -zo);
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3715,22 +3715,16 @@ public:
       // Octave-specific properties
       array_property __colormap__ hu , Matrix ()
       double_property mousewheelzoom , 0.5
 
       // hidden properties for alignment of subplots
       radio_property __autopos_tag__ h , "{none}|subplot"
       // hidden properties for inset
       array_property looseinset hu , Matrix (1, 4, 0.0)
-      // hidden properties for transformation computation
-      array_property x_viewtransform h , Matrix (4, 4, 0.0)
-      array_property x_projectiontransform h , Matrix (4, 4, 0.0)
-      array_property x_viewporttransform h , Matrix (4, 4, 0.0)
-      array_property x_normrendertransform h , Matrix (4, 4, 0.0)
-      array_property x_rendertransform h , Matrix (4, 4, 0.0)
       // hidden properties for minor ticks
       row_vector_property xminortickvalues h , Matrix ()
       row_vector_property yminortickvalues h , Matrix ()
       row_vector_property zminortickvalues h , Matrix ()
       // hidden property for text rendering
       double_property __fontsize_points__ hgr , 0
    END_PROPERTIES
 
